### ISSUE TITLE
OpenSSL 3.4.0 crashes on arm64_32 devices (for example, Apple Watch)

### ISSUE THREAD

**Issue by levlam** [openssl/openssl#26135]
`arm64_32` is the architecture that can be used on the modern arm64 Apple Watch devices with 32-bit pointer size. The crash happens on the first call to `ossl_ht_get`, which is called from `evp_generic_fetch`, making impossible any use of OpenSSL. The cause of the issue seems to be the function `apple_atomic_load_n_pvoid` introduced in the commit f5b5a35c84626823364b0c8535b968c106690a56, which is implemented in assembly for 64-bit Apple devices and expecting pointers to be 64-bit.

To reproduce the crash the code
```
#include <openssl/evp.h>

void test_pbkdf2() {
  unsigned char buf[32];
  unsigned char salt[4];
  salt[0] = salt[1] = salt[2] = salt[3] = 's';
  PKCS5_PBKDF2_HMAC("password", 8, salt, 4, 2, EVP_sha256(), 32, buf);
}
```
can be built for watchOS and run on a real device (emulators use host architecture and will not crash).

The issue was initially found by @vincentneo, who created an example project that reproduces the issue at https://github.com/vincentneo/OpenSSL340Crash.

Also, see https://github.com/tdlib/td/issues/3171#issuecomment-2525212205 for the exact position of the crash.

**Comment by t8m** [comment]
Is there something like `__aarch64_32__` defined on this platform?

**Comment by tom-cosgrove-arm** [comment]
Looking at https://reviews.llvm.org/D63131#change-RyFMReTm1DPr it appears possible that `__ARM64_ARCH_8_32__` might be defined (otherwise it's some horrible combination of `__APPLE__ && __aarch64__ && !__LP64__`), but I'm not set up to test this

**Comment by t8m** [comment]
Fix in #26145, please test it.

**Comment by vincentneo** [comment]
@t8m, based on the example project, utilising #25145 works fine on my Apple Watch.
